lms/djangoapps/courseware/access.py

```diff
@@ -173,53 +182,6 @@ def has_access(user, action, obj, course_key=None):
                     .format(type(obj)))
 
 
-# ================ Implementation helpers ================================
-
-def has_staff_access_to_preview_mode(user, obj, course_key=None):
```

Comment:
@muzaffaryousaf after this implementation, preview mode is working same as live mode except the fact that preview mode is not available for non-staff users. Is acceptance criteria covered here?
PR you mentioned says that preview mode will not be working for non-staff users. That is exactly achieved here with an extra feature that staff users will be able to masquerade as learner.